core/src/test/java/org/mskcc/oncotree/utils/TumorTypesUtilTest.java

```diff
@@ -155,12 +156,58 @@ public void testFullOncoTreeCodeListByCode() throws Exception {
+        for (String code : tumorTypeMap.keySet()) {
+            if (isSubtree) {
+                count = count + 1; //count current node if within subtree .. otherwise ignore root(s)
+            }
+            Map<String, TumorType> children = tumorTypeMap.get(code).getChildren();
+            if (children != null && children.keySet().size() > 0) {
+                count = count + countNonRootNodes(children, true);
+            }
+        }
+        return count;
+    }
+
     @Test
     public void testFullTumorTypesTxt() throws Exception {
         int failureCount = 0;
         StringBuilder failureReport  = new StringBuilder();
-        Map<String,TumorType> returnedTumorTypeMap = getReturnedFullTumorTypeMap();
-        //For this to work, children must be set:   InputStream tumorTypeTxtInputStream = tumorTypesUtil.getTumorTypeInputStream(returnedTumorTypeMap);
+        Map<String, TumorType> returnedTumorTypeMap = tumorTypesUtil.getTumorTypesByVersionFromRaw(mockVersion);
+        InputStream tumorTypeTxtInputStream = tumorTypesUtil.getTumorTypeInputStream(returnedTumorTypeMap);
+        BufferedReader returnedSheetReader = new BufferedReader(new InputStreamReader(tumorTypeTxtInputStream));
+        //check that header is present and matches
+        String header = returnedSheetReader.readLine();
+        if (!header.equals(TumorTypesUtil.TSV_HEADER)) {
+            failureReport.append("Header does not match what was expected, received : " + header + "\n\texpected:" + TumorTypesUtil.TSV_HEADER);
+            failureCount = failureCount + 1;
+        }
+        //check that each line is proper width and that line count matches the count of all nodes, minus root(s)
+        int expectedDataLineCount = countNonRootNodes(returnedTumorTypeMap, false);
+        int headerColumnCount = header.split("\t", -1).length;
+        int linesRead = 0;
+        while (returnedSheetReader.ready()) {
+            String nextLine = returnedSheetReader.readLine();
+            if (nextLine == null) {
+                break;
+            }
+            linesRead = linesRead + 1;
+            int lineColumnCount = nextLine.split("\t", -1).length;
+            if (lineColumnCount != headerColumnCount) {
+                failureReport.append("Line in TSV table has incorrect number of columns. expected: " + Integer.toString(headerColumnCount) +
+                        " seen: " + Integer.toString(lineColumnCount) + " on line:\n\t:" + nextLine);
+                failureCount = failureCount + 1;
+            }
+        }
+        if (expectedDataLineCount != linesRead) {
+            failureReport.append("Incorrect number of data lines. Expected: " + Integer.toString(expectedDataLineCount) + " received: " + Integer.toString(linesRead));
+            failureCount = failureCount + 1;
+        }
+        //TODO: add test for alphabetized order and the proper filling in of parents in primary, secondary, tertiary...; also meta fields
         if (failureCount > 0) {
```

Comment:
Just had a thought about this pattern thats being used throughout the oncotree tests: should we have multiple checks throughout a single test that append to a stringbuilder/counter, or have all these as separate tests that get called individually? You could have all the string parsing and stuff in some setup `@Before` method and the tests just operate on those. Not sure what the right answer is - what is currently being done is fine to me. @angelicaochoa, @n1zea144  maybe you can comment?